symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/ExtensionAppAuthenticatorRsaImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.symphony.bdk.core.auth.impl;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AppAuthSession;
+import com.symphony.bdk.core.auth.ExtensionAppAuthenticator;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.auth.jwt.JwtHelper;
+import com.symphony.bdk.gen.api.AuthenticationApi;
+import com.symphony.bdk.gen.api.model.AuthenticateExtensionAppRequest;
+import com.symphony.bdk.gen.api.model.ExtensionAppTokens;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.security.PrivateKey;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Extension app authenticator RSA implementation.
+ *
+ * @see <a href="https://developers.symphony.com/extension/docs/application-authentication#section-verifying-decoding-and-using-the-jwt">Application Authentication</a>
+ */
+@Slf4j
+public class ExtensionAppAuthenticatorRsaImpl implements ExtensionAppAuthenticator {
+
+  private final JwtHelper jwtHelper = new JwtHelper();
+
+  private final String appId;
+  private final PrivateKey appPrivateKey;
+  private final AuthenticationApi authenticationApi;
+
+  public ExtensionAppAuthenticatorRsaImpl(String appId, PrivateKey appPrivateKey, ApiClient loginApiClient) {
+    this.appId = appId;
+    this.appPrivateKey = appPrivateKey;
+    this.authenticationApi = new AuthenticationApi(loginApiClient);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Nonnull
+  @Override
+  public AppAuthSession authenticateExtensionApp(@Nonnull String appToken)
+      throws AuthUnauthorizedException {
+    AppAuthSession authSession = new AppAuthSessionRsaImpl(this, appToken);
+    authSession.refresh();
+    return authSession;
+  }
+
+  protected ExtensionAppTokens retrieveExtensionAppSession(String appToken) throws AuthUnauthorizedException {
+    log.debug("Start authenticating extension app with id : {} ...", this.appId);
+
+    final String jwt = this.jwtHelper.createSignedJwt(this.appId, 30_000, this.appPrivateKey);
```

Comment:
30_000 could be a constant (feels like a magic number here)

Reply:
I took this 30_000 from the other RSA authentication class. 😅 It should be a constant stored somewhere (JwtHelper I guess).
And in the legacy the value is 300_000, @symphony-thibault Can you confirm the value of expiration value of the jwt?

Reply:
Yes that's the right value : https://developers.symphony.com/symphony-developer/docs/rsa-bot-authentication-workflow#generate-a-signed-jwt